CMakeLists.txt

```diff
@@ -377,7 +377,12 @@ target_link_libraries(ufs PUBLIC "${_ufs_libs_public}")
 add_executable(ufs_model driver/UFS.F90)
 add_dependencies(ufs_model ufs)
 target_link_libraries(ufs_model ufs esmf w3emc::w3emc_d)
-set_target_properties(ufs_model PROPERTIES LINKER_LANGUAGE Fortran)
+
+if(CMAKE_Platform MATCHES "macosx.gnu")
+  set_target_properties(ufs_model PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
To suppress all those warnings, this additional line is working for me to pass the -w flag to the linker to just turn them off:
     target_link_options(ufs_model PRIVATE "-Wl,-w")
I also tried using -Wl,-no_compact_unwind mentioned in a comment instead of -Wl,-w. It suppressed the warnings, but the "key 'NUOPC' not found" error came back, as also noted in the comment.